packages/website/src/components/editor/LoadedEditor.tsx

```diff
@@ -56,33 +58,29 @@ export const LoadedEditor: React.FC<LoadedEditorProps> = ({
       // eslint-disable-next-line no-console
       console.info('[Editor] linting triggered');
 
-      const [markers, fatalMessage, codeActions] = lintCode(
-        webLinter,
-        code,
-        rules,
-        jsx,
-        sourceType,
-      );
-      fixes.clear();
-      for (const codeAction of codeActions) {
-        fixes.set(codeAction[0], codeAction[1]);
-      }
+      webLinter.updateParserOptions(jsx, sourceType);
+
+      const messages = webLinter.lint(code, rules ?? {});
+
+      const markers = parseLintResults(messages, codeActions);
 
       sandboxInstance.monaco.editor.setModelMarkers(
         sandboxInstance.editor.getModel()!,
         sandboxInstance.editor.getId(),
         markers,
       );
 
-      if (fatalMessage) {
-        setDecorations(
-          sandboxInstance.editor.deltaDecorations(decorations, []),
+      // fallback when event is not preset, ts < 4.0.5
+      if (!sandboxInstance.monaco.editor.onDidChangeMarkers) {
+        const markers = sandboxInstance.monaco.editor.getModelMarkers({});
+        onMarkersChange(
+          parseMarkers(markers, codeActions, sandboxInstance.editor),
```

Comment:
this is not ideal, as we in some cases are not going to retrieve typescript errors, but to my knowledge there is no alternative if we can't use `onDidChangeMarkers`

we may not see ts errors if monaco ts webworker is still processing data, as we are updating this only after linting is done